### PR TITLE
openjfx17: init at 17.0.0.1+1

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Add the OpenJFX17 to go along with OpenJDK17.
This is required to update cryptomator to a newer version.

It's currently only buildable on 64bit linux due to gradle >= 6 not supporting 32bit linux anymore.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/gradle/default.nix [hunk 1]
```diff
@@ -100,13 +100,17 @@ rec {
...
+    # Gradle 7 ships some binaires that are only available for some platforms
+    supportedPlatforms =  [ "x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin" "x86_64-cygwin" "x86_64-windows" "i686-windows"  ];
   };
 
   gradle_6 = gen {
     version = "6.9.2";
     nativeVersion = "0.22-milestone-20";
     sha256 = "13qyk3f6namw27ynh6nxljxpk9r3l12vxl3f0qpglprdf3c6ydcb";
     defaultJava = jdk11;
+    # Gradle 6 ships some binaires that are only available for some platforms
```
Comment (lorenzleutgeb) [inline pkgs/development/tools/build-managers/gradle/default.nix:112]:
```suggestion
    # Gradle 6 ships some binaries that are only available for some platforms.
```

File: pkgs/development/tools/build-managers/gradle/default.nix [hunk 2]
```diff
@@ -100,13 +100,17 @@ rec {
     nativeVersion = "0.22-milestone-21";
     sha256 = "00h3z0vxc4hv31sc71gb88r8yabyqgz304wpr0bxhbx2d14f11mm";
     defaultJava = jdk17;
+    # Gradle 7 ships some binaires that are only available for some platforms
```
Comment (lorenzleutgeb) [inline pkgs/development/tools/build-managers/gradle/default.nix:103]:
```suggestion
    # Gradle 7 ships some binaries that are only available for some platforms.
```

**Review by lorenzleutgeb — CHANGES_REQUESTED** [review]
> "ships some binaries that are only available for some platforms"

is a bit vague. Where do I have to look? How do I find out which architectures are supported?

File: pkgs/development/compilers/openjdk/openjfx/17.nix [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  });
+
+  makePackage = args: stdenv.mkDerivation ({
+    version = "${major}${update}${build}";
+
+    src = fetchFromGitHub {
+      owner = "openjdk";
+      repo = "jfx";
+      rev = repover;
+      sha256 = "sha256-PSiE9KbF/4u9VyBl9PAMLGzKyGFB86/XByeh7vhL6Kw=";
```
Comment (liff) [inline pkgs/development/compilers/openjdk/openjfx/17.nix:21]:
```suggestion
      hash = "sha256-PSiE9KbF/4u9VyBl9PAMLGzKyGFB86/XByeh7vhL6Kw=";
```

File: pkgs/development/compilers/openjdk/openjfx/17.nix [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        | sh
+      rm -rf $out/tmp
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    # Downloaded AWT jars differ by platform.
+    outputHash = {
+      x86_64-linux = "sha256-dV7/U5GpFxhI13smZ587C6cVE4FRNPY0zexZkYK4Yqo=";
+      i686-linux = "0hx69p2z96p7jbyq4r20jykkb8gx6r8q2cj7m30pldlsw3650bqa";
```
Comment (liff) [inline pkgs/development/compilers/openjdk/openjfx/17.nix:68]:
```suggestion
      i686-linux = "sha256-Ci9QzOCaNnrBqEcygVE2/aE1p5dAZIL9kuea9MVNpkM=";
```

from 

```shell
nix hash to-sri sha256:0hx69p2z96p7jbyq4r20jykkb8gx6r8q2cj7m30pldlsw3650bqa
```

File: pkgs/development/compilers/openjdk/openjfx/17.nix [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    installPhase = ''
+      find $GRADLE_USER_HOME -type f -regex '.*/modules.*\.\(jar\|pom\)' \
+        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
+        | sh
+      rm -rf $out/tmp
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    # Downloaded AWT jars differ by platform.
```
Comment (liff) [inline pkgs/development/compilers/openjdk/openjfx/17.nix:65]: No support for aarch64?

File: pkgs/development/tools/build-managers/gradle/default.nix (hunk 1 above)
Comment (bachp) [reply pkgs/development/tools/build-managers/gradle/default.nix]: Fixed

File: pkgs/development/compilers/openjdk/openjfx/17.nix (hunk 4 above)
Comment (bachp) [reply pkgs/development/compilers/openjdk/openjfx/17.nix]: This is acutally not needed anymore. i386 is no longer supported and aarch64 has the same hash as x86_64.

File: pkgs/development/compilers/openjdk/openjfx/17.nix (hunk 5 above)
Comment (bachp) [reply pkgs/development/compilers/openjdk/openjfx/17.nix]: aarch64 is indeed supported b ut has the same hash. I removed the conditional hash.

**Review by lorenzleutgeb — APPROVED** [review]
Changes to Gradle LGTM.

File: pkgs/development/tools/build-managers/gradle/default.nix (hunk 2 above)
Comment (bachp) [reply pkgs/development/tools/build-managers/gradle/default.nix]: I added a link to the supported platforms

**Comment by bachp** [comment]
Anything missing here? I need this to properly fix cryptomator in nixpkgs.

**Comment by bachp** [comment]
@abbradar please take a look

**Review by abbradar — APPROVED** [review]

**Pull request merged**